lib/hexdocs/queue.ex

```diff
@@ -250,4 +266,96 @@ defmodule Hexdocs.Queue do
       )
     end)
   end
+
+  defp find_search_items(package, version, files) do
+    search_data =
+      Enum.find_value(files, fn {path, content} ->
+        case Path.basename(path) do
+          "search_data-" <> _digest -> {:elixir, content}
+          "search-data.js" -> {:gleam, content}
+          _other -> nil
+        end
+      end)
+
+    case search_data do
+      {:elixir, content} ->
+        extract_elixir_search_items(package, version, content)
+
+      {:gleam, content} ->
+        extract_gleam_search_items(package, version, content)
```

Comment:
If we are going to index Gleam, I'd like explicit confirmation from them that they are willing to make search-data.js public and retain backwards compatibility. We should also double check that their data is ready for indexing. Previous ExDoc versions had a stripped down version of the docs (we would convert markdown to html and then strip the html) which sometimes would lead to suboptimal results. :)

Reply:
Maybe it is best if Gleam follows the same format as ExDoc, so we don't have to support multiple different formats over time? The one in ExDoc has been made generic over time exactly for cases like this.

Reply:
That's not a bad idea. I just pushed a proglang field to ExDoc, so they could adhere to that and set the proglang as "gleam".

Reply:
I think I'll revert the index-gleam commit, and return it once we either have a confirmation from the Gleam team that the format of their search_data is stable or they switch to ExDoc format.